packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts

```diff
@@ -0,0 +1,339 @@
+  headerText: 'ms-Panel-headerText',
+  content: 'ms-Panel-content',
+  footer: 'ms-Panel-footer',
+  footerInner: 'ms-Panel-footerInner',
+  isOpen: 'is-open',
+  hasCloseButton: 'ms-Panel--hasCloseButton',
+  smallFluid: 'ms-Panel--smFluid',
+  smallFixedNear: 'ms-Panel--smLeft',
+  smallFixedFar: 'ms-Panel--sm',
+  medium: 'ms-Panel--md',
+  large: 'ms-Panel--lg',
+  largeFixed: 'ms-Panel--fixed',
+  extraLarge: 'ms-Panel--xl',
+  custom: 'ms-Panel--custom'
+};
+
+function getIconStyles(props: IPanelStyleProps): IStyleFunctionOrObject<IIconStyleProps, IIconStyles> {
+  const { theme } = props;
+  return () => ({
+    root: {
+      height: 'auto',
+      width: '44px',
+      color: theme.palette.neutralSecondary,
+      fontSize: IconFontSizes.large
+    },
+    rootHovered: {
+      color: theme.palette.neutralPrimary
+    }
+  });
+}
+
+export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
+  const {
+    className,
+    focusTrapZoneClassName,
+    hasCloseButton,
+    headerClassName,
+    isAnimating,
+    isFooterAtBottom,
+    isFooterSticky,
+    isOnRightSide,
+    isOpen,
+    isHiddenOnDismiss,
+    theme,
+    type
+  } = props;
+  const { palette } = theme;
+
+  const panelSize = {
+    width: {
+      xs: '272px',
+      sm: '340px',
+      md: '643px',
+      lg: '940px'
+    },
+    margin: {
+      md: '48px',
+      lg: '428px',
+      xl: '176px'
+    }
+  };
+  const commandBarHeight = '44px';
+
+  const sharedPaddingStyles = {
+    paddingLeft: '16px',
+    paddingRight: '16px',
+    selectors: {
+      ['@media screen and (min-width: ' + ScreenWidthMinUhfMobile + 'px)']: {
+        paddingLeft: '32px',
+        paddingRight: '32px'
+      },
+      ['@media screen and (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
+        paddingLeft: '40px',
+        paddingRight: '40px'
+      }
+    }
+  };
+
+  const classNames = getGlobalClassNames(GlobalClassNames, theme);
+
+  return {
+    root: [
+      classNames.root,
+      isOpen && classNames.isOpen,
+      hasCloseButton && classNames.hasCloseButton,
+      {
+        pointerEvents: 'none',
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0
+      },
+      !isOpen &&
+        !isAnimating &&
+        isHiddenOnDismiss && {
+          visibility: 'hidden'
+        },
+      className
+    ],
+    overlay: [
+      {
+        pointerEvents: 'none',
+        opacity: 1,
+        cursor: 'pointer',
+        transition: `opacity ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
+      },
+      isOpen && {
+        cursor: 'pointer',
+        pointerEvents: 'auto'
+      },
+      isOpen && isAnimating && AnimationClassNames.fadeIn200,
+      !isOpen && isAnimating && AnimationClassNames.fadeOut200
+    ],
+    hiddenPanel: [
+      !isOpen &&
+        !isAnimating &&
+        isHiddenOnDismiss && {
+          visibility: 'hidden'
+          // visibility: 'visible'
```

Comment:
nit: this line can probably be deleted

Reply:
Ha, that's not a nit Kev, that's cleanup I missed :P